src/Nfts/ImageGallery/ImageGallery.tsx

```diff
@@ -58,81 +88,159 @@ interface ModeratedImageProps {
   }
 
   return (
-    <TouchableOpacity onPress={onPress} style={styles.imageContainer}>
+    <TouchableOpacity disabled={isImageBlocked} onPress={onPress} style={[styles.imageContainer]}>
       {isImageApproved ? (
-        <>
-          <Image source={{uri: image}} style={styles.image} />
-          <Spacer height={8} />
-          <Text style={styles.textTop}>{text}</Text>
-          <Spacer height={13} />
-        </>
-      ) : (
-        <>
-          <View style={styles.image} />
-          <Spacer height={8} />
-          <Text style={styles.textTop}>Image is not approved</Text>
-          <Spacer height={13} />
-        </>
-      )}
+        <ApprovedNFT text={text} uri={image} />
+      ) : isImageWithConsent ? (
+        <RequiresConsentNFT text={text} uri={image} />
+      ) : isImageBlocked ? (
+        <BlockedNFT text={text} />
+      ) : null}
     </TouchableOpacity>
   )
 }
 
-function SkeletonImagePlaceholder() {
+function BlockedNFT({text}: {text: string}) {
+  return (
+    <>
+      <Image source={placeholderImage} style={[styles.image, {width: imageSize, height: imageSize}]} />
+      <Spacer height={IMAGE_PADDING} />
+      <Text style={[styles.textTop, {width: imageSize}]}>{text}</Text>
+      <Spacer height={TEXT_PADDING} />
+    </>
+  )
+}
+
+function RequiresConsentNFT({uri, text}: {text: string; uri: string}) {
   return (
-    <SkeletonPlaceholder enabled={true}>
-      <View style={styles.imageContainer}>
-        <View style={styles.image} />
-        <SkeletonPlaceholder.Item style={styles.textTop} marginTop={8} marginBottom={13}>
-          <Text style={styles.textTop}>Loading...</Text>
-        </SkeletonPlaceholder.Item>
+    <View>
+      <View style={styles.imageWrapper}>
+        <Image source={{uri}} style={[styles.image, {width: imageSize, height: imageSize}]} blurRadius={20} />
+        <View style={styles.eyeWrapper}>
+          <Icon.EyeOff size={20} color="#FFFFFF" />
+        </View>
       </View>
-    </SkeletonPlaceholder>
+      <Spacer height={IMAGE_PADDING} />
+      <Text style={[styles.textTop, {width: imageSize}]}>{text}</Text>
+      <Spacer height={TEXT_PADDING} />
+    </View>
+  )
+}
+
+function ApprovedNFT({uri, text}: {text: string; uri: string}) {
+  return (
+    <>
+      <Image source={{uri}} style={[styles.image, {width: imageSize, height: imageSize}]} />
+      <Spacer height={IMAGE_PADDING} />
+      <Text style={[styles.textTop, {width: imageSize}]}>{text}</Text>
+      <Spacer height={TEXT_PADDING} />
+    </>
+  )
+}
+
+function SkeletonImagePlaceholder() {
+  return (
+    <View
+      style={[
+        styles.imageContainer,
+        {width: imageSize + 10, height: imageSize + IMAGE_PADDING + TEXT_SIZE + TEXT_PADDING},
+      ]}
+    >
+      <SkeletonPlaceholder enabled={true}>
+        <View>
+          <View style={{width: imageSize, height: imageSize, borderRadius: 8}} />
+          <View
+            style={{
+              marginTop: TEXT_PADDING,
```

Comment:
no margin

Reply:
I could not find any other way to make the `SkeletonPlaceholder` work. It worked only with margin